src/identity/caclient.rs

```diff
@@ -225,21 +229,8 @@ pub mod mock {
         }
 
         pub async fn set_error(&mut self, error: bool) {
-            if error {
-                let mut state = self.state.write().await;
-                state.fetches.push(Identity::Spiffe {
-                    trust_domain: "error".into(),
-                    namespace: "error".into(),
-                    service_account: "error".into(),
-                });
-            } else {
-                let mut state = self.state.write().await;
-                state.fetches.push(Identity::Spiffe {
-                    trust_domain: "success".into(),
-                    namespace: "success".into(),
-                    service_account: "success".into(),
-                });
-            }
+            let mut state = self.state.write().await;
```

Comment:
If we are going to skirt the use of proper mocks in this way, let's at least add conditional compilation to the test-only methods.

Reply:
This isn't skirting it; the old logic was literally not doing anything. It was supposed to be setting an error (so the mock would return an error) but actually just pushed it onto a list that was unused